File: 02-programacion-basica/09-arreglos-y-matrices/scala/ejemplos-guiados/ArreglosYMatrices.scala
object ArreglosYMatrices:
  def main(args: Array[String]): Unit =
    val notas = Array(12, 18, 15, 10)
    val promedio = notas.sum.toDouble / notas.length

    val matriz = Array(
      Array(1, 2),
      Array(3, 4)
    )

    println(s"Notas: ${notas.mkString("[", ", ", "]")}")
    println(s"Promedio: $promedio")
    println(s"Elemento fila 1, columna 0: ${matriz(1)(0)}")

File: 02-programacion-basica/09-arreglos-y-matrices/java/ejemplos-guiados/ArreglosYMatrices.java
import java.util.Arrays;

public class ArreglosYMatrices {
    public static void main(String[] args) {
        int[] notas = {12, 18, 15, 10};
        int suma = 0;

        // El for-each recorre todos los elementos sin manipular índices.
        for (int nota : notas) {
            suma += nota;
        }

        int[][] matriz = {{1, 2}, {3, 4}};
        double promedio = suma / (double) notas.length;

        System.out.println("Notas: " + Arrays.toString(notas));
        System.out.println("Promedio: " + promedio);
        System.out.println("Elemento fila 1, columna 0: " + matriz[1][0]);
    }
}
